src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+                    if (remainingCapacity > 0 && minNodeInMachineSet < maxDeveloperInstancesInCluster) {
+                        log.info("creating an instance");
+                        var payload = new KafkaRequestPayload()
+                            .name(INSTANCE_NAME_PREFIX + placeholderInstanceSuffixCounter.getAndIncrement() + "-" + Environment.LAUNCH_KEY)
+                            .plan(PLAN_STANDARD)
+                            .cloudProvider(Environment.CLOUD_PROVIDER)
+                            .region(Environment.DEFAULT_KAFKA_REGION);
+                        try {
+                            KafkaMgmtApiUtils.attemptCreatingKafkaInstance(kafkaMgmtApi, payload, Duration.ofSeconds(20), Duration.ofSeconds(20));
+                        } catch (KafkaClusterCapacityExhaustedException e) {
+                            log.warn("capacity exhausted at the moment %s", e);
+                        }
+                    }
+
+                    // observe if newly observed value in nodes increased from original snapshot
+                    Map<String, Integer> machineSetToReadyNodeCountSnapshotCurrent = FleetshardUtils.getReadyNodesPerEachMachineSetContainingName(oc, "kafka-standard");
+                    for (var currentlyObservedMachineSet  : machineSetToReadyNodeCountSnapshotCurrent.entrySet()) {
+                        // naming
+                        String machineSetName = currentlyObservedMachineSet.getKey();
+                        int newlyObservedValue = currentlyObservedMachineSet.getValue();
+                        int oldObservedValue = machineSetToReadyNodeCountSnapshotBefore.get(currentlyObservedMachineSet.getKey());
+
+                        // observing if value changed
+                        log.debug("machineSet '{}' capacity before '{}' and now '{}'", machineSetName, oldObservedValue, newlyObservedValue);
+                        if (newlyObservedValue > oldObservedValue) {
+                            log.info("MachineSet '{}' changed scaled number of its nodes", machineSetName);
+                            return true;
+                        }
+                    }
+
+                    // observe if any of newly crated kafka instance really is ready state (node for sure scaled), otherwise continue waiting
+                    return kafkaMgmtApi
+                        .getKafkas(null, null, null, null)
+                        .getItems().stream()
+                        .filter(e -> e.getName().contains(INSTANCE_NAME_PREFIX))
+                        .filter(e -> e.getOwner().equals(Environment.PRIMARY_USERNAME))
+                        .anyMatch(e -> e.getStatus().equals("ready"));
+                }
+            );
+
+        } catch (Exception ignored) {
+        } finally {
+            // delete and wait for cleaning of all instances spawned
+            KafkaMgmtApiUtils.deleteSearchedKafkaInstancesByOwner(kafkaMgmtApi, INSTANCE_NAME_PREFIX, Environment.PRIMARY_USERNAME);
```

Comment:
Will be good to test node downscaling as well.

i.e after the Kafka is deleted; In about 15 minutes there should be a downscaling happening as well. This can followup the PR.

Reply:
yeah i was observing that downscaling takes much longer than autoscaling, also if anybody will create new instances this process will be even more slowed down and I think that across scale of 15 minutes we would experience this in like 9/10 times.

Reply:
Yes, node downscaling is tricky to test and it also takes longer. I am okay with improving on this in future PR.

Reply:
okay there is a place for testing that (give timeout 15 minutes) waiting for downscaling, and making sure that no other instancess were deleted, if they were allow the difference between instances existed then and now to not be down from capacity field. 
e.g., 
**ideal case:** before test I had max capacity 7, and reached capacity 4
instances [a,b,c,d] as part of test I created instance  K,  and waited for test to observe increase in cap to 5. (assuming nothing was deleted) now i deleted instance K and querry capacity field while waiting it to fall back to 4 for next 15 min  at most. 

 **not so ideal case** before test I had max capacity 7, and reached capacity 4
instances [a,b,c,d]  i created instance K,  instances c,d were removed, and also instance e were created, 
am i expecting increase current reached capacity to (at most 15 minutes since i started to deleting instance K) to be 
at most on [a,b,c,d] which would give me 4 = 4 
-  deleted [c,d] plays no role as i would need to take into accout time of their deletion which could make this test possibly infinite
+ [e] which means 1 another instances (could be created the moment when i deleted instance K or could not: don't care), what i care about is that because i had original capacity 5 (from instances [a,b,c,d,K] now expecting it to go to 5 again because at worst case scenario it literally switches from [a,b,c,d,K] to [a,b,c,d,e]

**very very wrong case**
same as above case but there are also several extra instances created even in begging (i.e., observing increase in capacity).

Reply:
There is also the case of the reserved instance standard. This manifest itself when you've:
- max capacity of 7
- capacity reached 6
- only one instance is created - no scale up happens as node count is already at its max (21 nodes)
- only one instance is deleted - no scale down happens (as the reserved instance will take up the space)

Another variation of this:

- max capacity of 7
- capacity reached 6
- only one instance is created - no scale up happens as node count is already at its max (21 nodes)
- several instances are deleted - node downscaling should  happen

Reply:
in scaling up these are covered as at worst case scenario we at least expect the instance to be actually created.

Reply:
so would you say that i put some extra logic for downscaling there ?

Reply:
In my opinion, we could just add a TODO and defer the logic for testing node downscaling in a followup PR.

Reply:
good good.